pkg/clients/hostedzone/hostedzone.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hostedzone
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/crossplane/provider-aws/apis/route53/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+// Client defines Route53 Client operations
+type Client interface {
+	CreateHostedZoneRequest(input *route53.CreateHostedZoneInput) route53.CreateHostedZoneRequest
+	DeleteHostedZoneRequest(input *route53.DeleteHostedZoneInput) route53.DeleteHostedZoneRequest
+	GetHostedZoneRequest(input *route53.GetHostedZoneInput) route53.GetHostedZoneRequest
+	UpdateHostedZoneCommentRequest(input *route53.UpdateHostedZoneCommentInput) route53.UpdateHostedZoneCommentRequest
+}
+
+// NewClient creates new AWS Client with provided AWS Configurations/Credentials
+func NewClient(config *aws.Config) Client {
+	return route53.New(*config)
+}
+
+// IsErrorNoSuchHostedZone returns true if the error code indicates that the requested Zone was not found
+func IsErrorNoSuchHostedZone(err error) bool {
+	if zoneErr, ok := err.(awserr.Error); ok && zoneErr.Code() == route53.ErrCodeNoSuchHostedZone {
+		return true
+	}
+	return false
+}
+
+// IsUpToDate check whether the comment in Spec and Response are same or not
+func IsUpToDate(cfg *v1alpha1.HostedZoneConfig, res *route53.HostedZoneConfig) bool {
+	if cfg == nil && res.Comment != nil {
+		return false
+	}
+	if cfg != nil && !cmp.Equal(aws.StringValue(cfg.Comment), aws.StringValue(res.Comment)) {
+		return false
+	}
+	return true
```

Comment:
Suggested change has been incorporated.

Reply:
But this doesn't makes much sense to me. Because `res.PrivateZone` is auto populated to `false` as the default by AWS. So technically `res` will never be `nil`. Also, if we have not provided either of the `cfg.Comment` or `cfg.PrivateZone` it will be `nil` and will return `false` which is not the intended behavior.